safehttp/plugins/xsrf/xsrf_test.go

```diff
@@ -163,87 +91,204 @@ func TestXSRFTokenMultipart(t *testing.T) {
+				req.Header.Set("Content-Type", "application/x-www-form-urlencoded")
+				return req
+			}(),
+		},
+		{
+			name: "Missing token in PATCH request with form",
+			req: func() *safehttp.IncomingRequest {
+				req := safehttptest.NewRequest(safehttp.MethodPatch, "/", strings.NewReader("foo=bar"))
 				req.Header.Set("Content-Type", "application/x-www-form-urlencoded")
 				return req
 			}(),
-			wantStatus: 401,
-			wantHeader: map[string][]string{
-				"Content-Type":           {"text/plain; charset=utf-8"},
-				"X-Content-Type-Options": {"nosniff"},
-			},
-			wantBody: "Unauthorized\n",
 		},
 		{
-			name: "Missing token in multipart request",
+			name: "Missing token in POST request with multipart form",
 			req: func() *safehttp.IncomingRequest {
 				b := "--123\r\n" +
 					"Content-Disposition: form-data; name=\"foo\"\r\n" +
 					"\r\n" +
 					"bar\r\n" +
 					"--123--\r\n"
-				req := safehttptest.NewRequest("POST", "/", strings.NewReader(b))
+				req := safehttptest.NewRequest(safehttp.MethodPost, "/", strings.NewReader(b))
+				req.Header.Set("Content-Type", `multipart/form-data; boundary="123"`)
+				return req
+			}(),
+		},
+		{
+			name: "Missing token in PATCH request with multipart form",
+			req: func() *safehttp.IncomingRequest {
+				b := "--123\r\n" +
+					"Content-Disposition: form-data; name=\"foo\"\r\n" +
+					"\r\n" +
+					"bar\r\n" +
+					"--123--\r\n"
+				req := safehttptest.NewRequest(safehttp.MethodPatch, "/", strings.NewReader(b))
 				req.Header.Set("Content-Type", `multipart/form-data; boundary="123"`)
 				return req
 			}(),
-			wantStatus: 401,
-			wantHeader: map[string][]string{
-				"Content-Type":           {"text/plain; charset=utf-8"},
-				"X-Content-Type-Options": {"nosniff"},
-			},
-			wantBody: "Unauthorized\n",
 		},
 	}
 	for _, test := range tests {
 		i := xsrf.Interceptor{AppKey: "xsrf", Identifier: userIdentifier{}}
 		rec := safehttptest.NewResponseRecorder()
+
 		i.Before(rec.ResponseWriter, test.req)
 
-		if rec.Status() != test.wantStatus {
-			t.Errorf("response status: got %v, want %v", rec.Status(), test.wantStatus)
+		if want, got := 401, rec.Status(); got != want {
+			t.Errorf("response status: got %v, want %v", got, want)
 		}
-		if diff := cmp.Diff(test.wantHeader, map[string][]string(rec.Header())); diff != "" {
+		wantHeaders := map[string][]string{
+			"Content-Type":           {"text/plain; charset=utf-8"},
+			"X-Content-Type-Options": {"nosniff"},
+		}
+		if diff := cmp.Diff(wantHeaders, map[string][]string(rec.Header())); diff != "" {
 			t.Errorf("rw.header mismatch (-want +got):\n%s", diff)
 		}
-		if got := rec.Body(); got != test.wantBody {
-			t.Errorf("response body: got %q want %q", got, test.wantBody)
+		if want, got := "Unauthorized\n", rec.Body(); got != want {
+			t.Errorf("response body: got %q want %q", got, want)
 		}
 	}
 }
+
+func TestTokenRequestContext(t *testing.T) {
```

Comment:
It feels like this test is testing `xsrftoken.Valid` and not `xsrf.Token`. We are already testing `xsrftoken.Valid` in `TestTokenPost` and `TestTokenMultipart`. I would suggest replacing this with two small tests. One which creates a `IncomingRequest`, places something in the context and then retrieves it with `xsrf.Token`. And another test in which you send a request to `Before`, just like you do in this test and checks afterwards if the `IncomingRequest` contains the token, without actually validating the token.